### PR TITLE
[GSSDLC-165] Sizing Import: Improve error feedback.

### PR DESCRIPTION
**Pull request by lohandus**

### CODE REVIEW

File: src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+                    "Please check the configuration of selected project in Jira.");
+        }
+        
+        return ValidationResult.success();
+    }
+    
+    private ValidationResult validateSpreadsheetExistence(ValidationContext context) {
+        try {
+            context.sheetsTitles = context.spreadsheetsManager.getSheetsTitles(context.spreadsheetId);
+        } catch (SpreadsheeNotFoundException ex) {
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java]: CRITICAL Either log or rethrow this exception. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1166)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'

File: src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+            return ValidationResult.success();
+
+        return ValidationResult.fail(
+                "Invalid spreadsheet format: Missing required columns.", 
+                missingColumns.stream()
+                    .map(c -> format("“%s” column should be placed at position “%s”", c.getName(), c.getColumnLetter()))
+                    .collect(joining(", ")) + ".");
+    }
+    
+    private ValidationResult validateStaticColumnsSingularity(ValidationContext context) {
```
Comment (taksan) [inline src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java]: Acho que um nome melhor para esse método seria "validateStaticColumnsUniqueness"

File: src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                "Invalid spreadsheet format: Row %s of sheet “%s” should contain the headers (e.g. %s).", 
+                headerRowNumber, SHEET_TITLE, expectedHeaders);
+        
+        String detail = format("Data to import should start at row %s.", config.getDataStartingRowNumber());
+        
+        return ValidationResult.fail(message, detail);
+    }
+    
+    private ValidationResult validateStaticColumnsExistence(ValidationContext context) {
+        context.staticColumnsOccurenceReport = sheetStaticColumns.get().stream()
```
Comment (taksan) [inline src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java]: Novamente, não seria melhor fazer esse setup do contexto num ponto de setup?

File: src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        int headersRowIndex = config.getDataStartingRowIndex() - 1;
+        SpreadsheetsManager spreadsheetsManager = googleApiService.buildSpreadsheetsManager();
+        ValidationContext context = new ValidationContext(projectKey, spreadsheetId, headersRowIndex, spreadsheetsManager);
+
+        Optional<ValidationResult> failedResult = validationSteps.stream()
+                .map(step -> step.validate(context))
+                .filter(ValidationResult::failed)
+                .findFirst();
+
+        return failedResult.orElse(ValidationResult.success());
```
Comment (taksan) [inline src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java]: Isso significa que se houverem múltiplas falhas de validação, apenas a primeira será mostrada né? 
Será que faria sentido mostra um report com todas as violações?

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+        
+        return validateSpreadsheetExistence(context);
+    }
+    
+    private ValidationResult validateSpreadsheetExistence(ValidationContext context) {
+        List<String> sheetsTitles;
+
+        try {
+            sheetsTitles = context.spreadsheetsManager.getSheetsTitles(context.spreadsheetId);
+        } catch (SpreadsheeNotFoundException ex) {
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/sizingImport/SizingImportValidator.java]: MAJOR Either log or rethrow this exception. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1166)

**Pull request merged**